src/components/KInput/KInput.vue

```diff
@@ -319,196 +238,148 @@ export default {
+  &:has(.k-input:disabled) {
+    cursor: not-allowed;
+  }
+  // workaround for Firefox
+  &.input-disabled {
+    cursor: not-allowed;
+  }
+
+  // error styles
+  &.input-error {
+    .k-input {
+      box-shadow: var(--kui-shadow-border-danger, $kui-shadow-border-danger);
 
-    &.icon-end {
-      padding-right: calc($kInputMediumSizingX + var(--kui-space-40, $kui-space-40) + $kInputMediumIconSize) !important; // account for icon offset and width
-      ~ .input-icon {
-        right: $kInputMediumSizingX;
+      // TODO: do we need a hover state for error?
+
+      &:focus {
+        box-shadow: var(--kui-shadow-border-danger, $kui-shadow-border-danger), var(--kui-shadow-focus, $kui-shadow-focus);
       }
     }
 
-    // input size small
-    $kInputSmallSizingY: var(--kui-space-40, $kui-space-40);
-    $kInputSmallSizingX: var(--kui-space-50, $kui-space-50);
-    $kInputSmallIconSize: var(--kui-icon-size-40, $kui-icon-size-40);
+    .help-text {
+      color: var(--kui-color-text-danger, $kui-color-text-danger);
+    }
+  }
 
-    &.k-input-small {
-      ~ .input-icon {
-        // (height of entire element (paddings + line height) - icon size) / 2
-        top: calc((($kInputSmallSizingY + $kInputSmallSizingY + $kInputLineHeight) - $kInputSmallIconSize) / 2);
+  .help-text {
+    color: var(--kui-color-text-neutral, $kui-color-text-neutral);
+    font-family: var(--kui-font-family-text, $kui-font-family-text);
+    font-size: var(--kui-font-size-20, $kui-font-size-20);
+    font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular);
+    line-height: var(--kui-line-height-20, $kui-line-height-20);
+    margin: 0;
+    margin-top: var(--kui-space-40, $kui-space-40);
+  }
 
-        :deep(svg) {
-          height: $kInputSmallIconSize;
-          width: $kInputSmallIconSize;
+  // slots styles
+  .input-element-wrapper {
+    position: relative;
+
+    .before-content-wrapper, .after-content-wrapper {
+      position: absolute;
+      top: 50%;
+      transform: translateY(-50%);
+
+      :deep(>) {
+        &:is(.kui-icon) {
+          color: var(--kui-color-text-neutral, $kui-color-text-neutral) !important;
+          /* stylelint-disable-next-line @kong/design-tokens/use-proper-token */
+          height: $kInputIconSize;
+          /* stylelint-disable-next-line @kong/design-tokens/use-proper-token */
+          width: $kInputIconSize;
         }
-      }
 
-      &.icon-start {
-        padding-left: calc($kInputSmallSizingX + var(--kui-space-40, $kui-space-40) + $kInputSmallIconSize) !important; // account for icon offset and width
-        ~ .input-icon {
-          left: $kInputSmallSizingX;
-        }
-      }
-      &.icon-end {
-        padding-right: calc($kInputSmallSizingX + var(--kui-space-40, $kui-space-40) + $kInputSmallIconSize) !important; // account for icon offset and width
-        ~ .input-icon {
-          right: $kInputSmallSizingX;
+        &:not(.kui-icon) {
+          display: none;
```

Comment:
issue: I don't think we should be opinionated on what developers slot in. It's up to them to not break it; the component should be "stupid"

Also, this directly ties the logic to a class in a different package that could change over time

Reply:
I guess you're right, removing this.

Reply:
Though we won't escape using a class from a different package because we do want to style the icon size from within here using `.kui-icon` class

Reply:
>  we do want to style the icon size from within here
You could generally target an SVG, but not the specific class

Reply:
That won't work because icons exported by our library are wrapped by the span element. We could target the psan element but I feel like that's wrong.

![Screen Shot 2023-10-11 at 10 28 57 AM](https://github.com/Kong/kongponents/assets/36751160/918e1511-8c54-4386-ba9d-df5d8b6b53ed)

Reply:
For the color, as long as you set the button `color` properly, the icon would just inherit the color since it defaults to `currentColor`

For the sizing, I'm wondering if you could just provide the size of the slot in general instead of targeting an element.

e.g. make a note that the slots are specifically in place to slot icons inside of the input, and that any slot content has a width/height rule in place of `Npx`

Reply:
Well yeah that works. But what if they decide to use KButton to make that icon clickable? I think we need to discuss this during our sync later today

Reply:
I'm not sure I understand what you mean; can demo in our meeting later 👍🏼